org.eclipse.paho.android.service/src/main/java/org/eclipse/paho/android/service/MqttAndroidClient.java

```diff
@@ -1146,7 +1146,7 @@ public IMqttToken subscribe(String[] topicFilters, int[] qos, Object userContext
 		String activityToken = storeToken(token);
 		mqttService.subscribe(clientHandle, topicFilters, qos, null, activityToken, messageListeners);
 
-		return null;
+		return token;

```

Comment:
It's already here https://github.com/hannesa2/paho.mqtt.android/blob/master/org.eclipse.paho.android.service/src/main/java/org/eclipse/paho/android/service/MqttAndroidClient.java#L960-L967

`implementation 'com.github.hannesa2:paho.mqtt.android:2.0'`

Reply:
Hi Hannes, I saw your solution and tested it, but unfortunately it didn't help me. 

I make a call to this method:
```java
client.subscribe(topic, qos, new IMqttMessageListener() {
            @Override
            public void messageArrived(String topic, MqttMessage message) throws Exception {
                Log.i(topic, message.toString());
            }
 });
```
When publishing messages in, nothing happens.

However, the message sent from the device under test appears after the screen is turned off and on. But other messages don't go through like that.

Reply:
[Link to my test project](https://github.com/mofrison/mqtt-trouble)